lib/zone.ts

```diff
@@ -1352,13 +1363,18 @@ const Zone: ZoneType = (function(global: any) {
           }
         }
       }
-      error.stack = error.zoneAwareStack = frames.join('\n');
+      if (this && this !== global) {
+        this.stack = this.zoneAwareStack = frames.join('\n');
+      } else {
+        error.stack = error.zoneAwareStack = frames.join('\n');
+        error.originalStack = originalStack;
+      }
     }
-    return error;
-  }
+    return (this && this !== global) ? this : error;
+  };
 
   // Copy the prototype so that instanceof operator works as expected
-  ZoneAwareError.prototype = NativeError.prototype;
+  ZoneAwareError.prototype = Object.create(NativeError.prototype);
```

Comment:
I just want to create a new object to let ZoneAwareError.prototype be different with NativeError's, so in the future, if someday modify ZoneAwareError's prototype will not impact NativeError's, for this issue #546, we don't need to use Object.create